sources/api/pluto/src/main.rs

```diff
@@ -270,11 +271,80 @@ fn pause_container_uri(region: &str) -> Result<String> {
     ))
 }
 
+/// Returns the number of CPUs on current instance.
+fn get_num_cpus() -> usize {
+    num_cpus::get()
+}
+
+/// Calculates the amount of memory to reserve for kubeReserved in mebibytes.
+/// This function is taken from EKS' calculations.
+/// https://github.com/awslabs/amazon-eks-ami/blob/db28da15d2b696bc08ac3aacc9675694f4a69933/files/bootstrap.sh#L228-L239
+fn get_memory_mebibytes_to_reserve(client: &Client, session_token: &str) -> Result<String> {
+    let max_num_pods = get_max_pods(&client, &session_token)?;
```

Comment:
A few issues here:
* others have mentioned the magic number, we need a comment explaining it.
* This results in a second query to IMDS, which is unfortunate.
* Doing this in `pluto` ties it to AWS.

Ideally we would update the memory reservation whenever `max-pods` was updated, or whenever the physical resources on the machine changed.

What would be really great here is if in the kubelet-config template, we could do something like this:
```
kubeReserved:
  {{~#if settings.kubernetes.kube-reserved.cpu}}
  cpu: {{settings.kubernetes.kube-reserved.cpu}}
  {{~else}
  cpu: {{calculate-cpu-reserve}}
  {{~/if}}
  {{~#if settings.kubernetes.kube-reserved.memory}}
  memory: {{settings.kubernetes.kube-reserved.memory}}
  {{~else}
  {{~#if settings.kubernetes.max-pods}}
  memory: {{calculate-mem-reserve settings.kubernetes.max-pods}}
  {{~/if}}
  {{~/if}}
```

Then if we were given a specific value, we'd always use it, but otherwise we'd run the calculation when the template was rendered.